src/System.Globalization/tests/NumberFormatInfo/NumberFormatInfoData.cs

```diff
@@ -93,5 +97,22 @@ internal static int[] GetCurrencyNegativePatterns(CultureInfo cultureInfo)
 
             throw DateTimeFormatInfoData.GetCultureNotSupportedException(cultureInfo);
         }
+        
+        private static Version GetOSXKernelVersion()
+        {
+            if (s_isOSX)
+            {
+                ulong bytesLength = 256;
+                byte[] bytes = new byte[bytesLength];
+                sysctlbyname("kern.osrelease", bytes, ref bytesLength, null, 0);
+                string versionString = Encoding.UTF8.GetString(bytes);
+                return Version.Parse(versionString);
+            }
+            
+            return new Version(0, 0, 0);
+        }
+        
+        [DllImport("libc")]
+        private static extern int sysctlbyname(string ctlName, byte[] oldp, ref ulong oldpLen, byte[] newp, ulong newpLen);
```

Comment:
> by the way I wrote a test on Windows which do what I am saying and covering much more areas and was reliable when running on different Windows versions.

Reliable, sure. Accurate in detecting failure, who knows ;)

Anyway, I've made my points, and I know you understand them.  It seems we just fundamentally disagree.

I believe we need to expose OS version information.  We're not doing so out of fear that developers may misuse it.  So be it.  If we don't provide it, developers are going to find worse, less reliable, less efficient ways to get at it, making the end result worse than if we'd just provided it in the first place.